test/conformance_test.go

```diff
@@ -38,33 +42,119 @@ func TestTaskRun(t *testing.T) {
 	knativetest.CleanupOnInterrupt(func() { tearDown(ctx, t, c, namespace) }, t.Logf)
 	defer tearDown(ctx, t, c, namespace)
 
-	trName := "echo-hello-task-run"
-	tr := &v1beta1.TaskRun{
-		ObjectMeta: metav1.ObjectMeta{Name: trName, Namespace: namespace},
-		Spec: v1beta1.TaskRunSpec{
-			TaskSpec: &v1beta1.TaskSpec{
-				Steps: []v1beta1.Step{{Container: corev1.Container{
-					Image:   "busybox",
-					Command: []string{"echo", "\"hello\""},
-				}}},
+	for _, tc := range []struct {
+		name                    string
+		trName                  string
+		tr                      *v1beta1.TaskRun
+		fn                      conditionFn
+		expectedConditionStatus corev1.ConditionStatus
+		expectedStepState       []v1beta1.StepState
+	}{{
+		name:   "successful-task-run",
+		trName: "echo-hello-task-run",
+		tr: &v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{Name: "echo-hello-task-run", Namespace: namespace},
+			Spec: v1beta1.TaskRunSpec{
+				TaskSpec: &v1beta1.TaskSpec{
+					Steps: []v1beta1.Step{{Container: corev1.Container{
+						Image:   "busybox",
+						Command: []string{"echo", "\"hello\""},
+					}}},
+				},
 			},
 		},
-	}
+		fn:                      TaskRunSucceed,
+		expectedConditionStatus: corev1.ConditionTrue,
+		expectedStepState: []v1beta1.StepState{{
+			ContainerState: corev1.ContainerState{
+				Terminated: &corev1.ContainerStateTerminated{
+					ExitCode: 0,
+					Reason:   "Completed",
+				},
+			},
+			Name:          "unnamed-0",
```

Comment:
So the [API spec](https://github.com/tektoncd/pipeline/blob/master/docs/api-spec.md#stepstate) doesn't specify whether implementations have to populate `Name` or `ContainerName`, or how should populate them when the user's requested step wasn't given a name, so we shouldn't require them in this test.

That being said, these seem like useful things to include in future changes to the spec, and cover in conformance tests. But for now, we should ignore them.

Reply:
I see. But I'm going to reserve the StepState struct although we only check one field of it for now. It might be used in the future.